auth/policies.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package auth
+
+import (
+	"context"
+
+	"github.com/mainflux/mainflux/pkg/errors"
+	acl "github.com/ory/keto/proto/ory/keto/acl/v1alpha1"
+)
+
+// PolicyReq represents an argument struct for making a policy related
+// function calls.
+type PolicyReq struct {
+	Subject  string
+	Object   string
+	Relation string
+}
+
+// Authz represents a authorization service. It exposes
+// functionalities through `auth` to perform authorization.
+type Authz interface {
+	// Authorize checks authorization of the given `subject`. Basically,
+	// Authorize verifies that Is `subject` allowed to `relation` on
+	// `object`. Authorize returns a non-nil error if the subject has
+	// no relation on the object (which simply means the operation is
+	// denied).
+	Authorize(ctx context.Context, pr PolicyReq) error
+
+	// AddPolicy creates a policy for the given subject. So that, after
+	// AddPolicy, `subject`has a `relation` on `object`. Returns non-nil
+	// error in case of failures.
+	AddPolicy(ctx context.Context, pr PolicyReq) error
+}
+
+// PolicyAgent facilitates the communication to authorization
+// services and implements Authz functionalities for certain
+// authorization services (e.g. ORY Keto).
+type PolicyAgent interface {
+	// CheckPolicy checks if the subject has a relation on the object.
+	// It returns a non-nil error if the subject has no relation on
+	// the object (which simply means the operation is denied).
+	CheckPolicy(ctx context.Context, pr PolicyReq) error
+
+	// AddPolicy creates a policy for the given subject. So that, after
+	// AddPolicy, `subject` has a `relation` on `object`. Returns non-nil
+	// error in case of failures.
+	AddPolicy(ctx context.Context, pr PolicyReq) error
+}
+
+type policyAgent struct {
```

Comment:
Please move implementation to a separate `keto` package, similar to the repository and `postgres`. We keep only the service interface and implementation together because there is no better place for implementation.